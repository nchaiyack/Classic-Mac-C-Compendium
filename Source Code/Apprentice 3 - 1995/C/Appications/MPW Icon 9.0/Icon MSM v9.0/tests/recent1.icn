procedure main ()   sf([])   write(args(main))   write(args(write))# show results of bitwise operations on various operand combinations   every i := 1 | '2' | "3" do {      write (       "    i        j       ~j      i & j    i | j    i ^ j   i << j   i >> j")      every j := 0 | 1 | 2 | 3 | 4 | 100 do {         write(right(i,8), right(j,9))         word (i)         word (j)         word (icom (j))         word (iand (i, j))         word (ior (i, j))         word (ixor (i, j))         word (ishift (i, j))         word (ishift (i, -j))         write ()         }      }# test remove() and rename(), and print errors in case of malfunction   name1 := "temp1"   name2 := "temp2"   data := "Here's the data"   every remove (name1 | name2)		# just in case   open (name1) & stop ("can't remove ", name1, " to initialize test")   open (name2) & stop ("can't remove ", name2, " to initialize test")   remove (name1) & stop ("successfully removed nonexistent file")   rename (name1, name2) & stop ("successfully renamed nonexistent file")   f := open (name1, "w") | stop ("can't open ",name1," for write")   write (f, data)   close (f)   f := open (name1) | stop ("can't open ",name1," after write")   s := read (f) | ""   close(f)   s == data | stop ("data lost after write")   rename (name1, name2) | stop ("can't rename(",name1,",",name2,")")   f := open (name2) | stop ("can't open ",name2," after rename")   s := read (f) | ""   close(f)   s == data | stop ("data lost after rename")   remove (name1) & stop ("remove succeeded on file already renamed")   remove (name2) | stop ("can't remove renamed file")   open (name1) & stop (name1, " still around at end of test")   open (name2) & stop (name2, " still around at end of test")#  test seek() and where()   f := open("concord.dat")   write(image(seek(f,11)))   write(where(f))   write(image(reads(f,10)))   write(where(f))   write(where(f))   seek(f,-2)   write(where(f))   write(image(reads(f,1)))   write(where(f))end# write word in hexadecimalprocedure word (v)   xd (v, 8)   writes (" ")   return   end# write n low-order hex digits of vprocedure xd (v, n)   xd (ishift (v, -4), 0 < n - 1)   writes ("0123456789ABCDEF" [1 + iand (v, 16r0F)])   return   end# ferr(func,val,err) -- call func(val) and verify that error "err" is producedprocedure ferr (func, val, err)   write(msg := "oops -- " || image(func) || "(" || image (val) || ") ")   returnendprocedure p(a, b, c[])   write("   image(a):", image(a))   write("   image(b):", image(b))   write("   image(c):", image(c))   write("   every write(\"\\t\", !c):")   every write("\t", !c)endprocedure q(a[])   write("   every write(\"\\t\", !a):")   every write("\t", !a)endprocedure show(t)   local x   write("   *t --> ", *t)   write("   t[\"xyz\"] --> ", image(t["xyz"]) | "failure")   write("   member(t, \"xyz\") --> ", image(member(t, "xyz")) | "failure")   x := sort(t, 3)   write("   contents of t:")   while writes("\t", image(get(x)), " : ")      do write(image(get(x)))   write("")end#  test the new sortf(x,n) functionglobal datarecord r1(a)record r3(a,b,c)procedure sf (args)    local n, z    z := []    every put (z, 1 to 100)    data := [       r3(3,1,4),       [1,5,9],       r3(2,6,5),       r3(3,5),       r1(2),       3,       r1(4),       r1(8),       [5,&null,5],       [4,4,4,4],       [3,3,3],       [&null,25],       4,       [2,2],       [1],       [&null,&null],       [],       r3(7,8,9),       z]    dump ("sort(L)", sort (data))    if *args = 0 then	every test (&null | 1 | "2" | '3' | 4 | 17 | -4 | -3 | "-2" | -1)    else	every test (!args)    endprocedure test (n)    local r1, r2    write ()    write ("-------------------- testing n = ", \n | "&null")    r1 := sortf (data, n)    r2 := sortf (set(data), n)    dump ("sortf(L,n)", r1)    if same (r1, r2) then	write ("\nsortf(S,n) [same]")    else	dump ("sortf(S,n) [********** OOPS -- results differ: **********]", r2)    endprocedure dump (s, l)    local e    write ()    write (s, ":")    every e := !l do {       writes ("  ", left(type(e), 8))       if (type(e) == ("r1" | "r3" | "list")) then	  every writes (" ", image(e[(1 to 5) | (95 to 100)]) | "\n")       else	  write (" ", image(e))       }    return    endprocedure same (a, b)    local i    if *a ~= *b then fail    every i := 1 to *a do	if a[i] ~=== b[i] then fail    return    end