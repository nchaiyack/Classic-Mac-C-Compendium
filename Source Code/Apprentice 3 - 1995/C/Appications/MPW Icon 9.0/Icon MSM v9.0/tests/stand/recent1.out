
sort(L):
  integer  3
  integer  4
  list     1 2 3 4 5 95 96 97 98 99 100 
  list     1 5 9 
  list     5 &null 5 
  list     4 4 4 4 
  list     3 3 3 
  list     &null 25 
  list     2 2 
  list     1 
  list     &null &null 
  list     
  r1       2 
  r1       4 
  r1       8 
  r3       3 1 4 
  r3       2 6 5 
  r3       3 5 &null 
  r3       7 8 9 

-------------------- testing n = &null

sortf(L,n):
  integer  3
  integer  4
  list     
  list     &null 25 
  list     &null &null 
  list     1 2 3 4 5 95 96 97 98 99 100 
  list     1 5 9 
  list     1 
  list     2 2 
  list     3 3 3 
  list     4 4 4 4 
  list     5 &null 5 
  r1       2 
  r3       2 6 5 
  r3       3 1 4 
  r3       3 5 &null 
  r1       4 
  r3       7 8 9 
  r1       8 

sortf(S,n) [same]

-------------------- testing n = 1

sortf(L,n):
  integer  3
  integer  4
  list     
  list     &null 25 
  list     &null &null 
  list     1 2 3 4 5 95 96 97 98 99 100 
  list     1 5 9 
  list     1 
  list     2 2 
  list     3 3 3 
  list     4 4 4 4 
  list     5 &null 5 
  r1       2 
  r3       2 6 5 
  r3       3 1 4 
  r3       3 5 &null 
  r1       4 
  r3       7 8 9 
  r1       8 

sortf(S,n) [same]

-------------------- testing n = 2

sortf(L,n):
  integer  3
  integer  4
  list     1 
  list     
  list     5 &null 5 
  list     &null &null 
  list     1 2 3 4 5 95 96 97 98 99 100 
  list     2 2 
  list     3 3 3 
  list     4 4 4 4 
  list     1 5 9 
  list     &null 25 
  r1       2 
  r1       4 
  r1       8 
  r3       3 1 4 
  r3       3 5 &null 
  r3       2 6 5 
  r3       7 8 9 

sortf(S,n) [same]

-------------------- testing n = 3

sortf(L,n):
  integer  3
  integer  4
  list     &null 25 
  list     2 2 
  list     1 
  list     &null &null 
  list     
  list     1 2 3 4 5 95 96 97 98 99 100 
  list     3 3 3 
  list     4 4 4 4 
  list     5 &null 5 
  list     1 5 9 
  r1       2 
  r1       4 
  r1       8 
  r3       3 5 &null 
  r3       3 1 4 
  r3       2 6 5 
  r3       7 8 9 

sortf(S,n) [same]

-------------------- testing n = 4

sortf(L,n):
  integer  3
  integer  4
  list     1 5 9 
  list     5 &null 5 
  list     3 3 3 
  list     &null 25 
  list     2 2 
  list     1 
  list     &null &null 
  list     
  list     1 2 3 4 5 95 96 97 98 99 100 
  list     4 4 4 4 
  r1       2 
  r1       4 
  r1       8 
  r3       3 1 4 
  r3       2 6 5 
  r3       3 5 &null 
  r3       7 8 9 

sortf(S,n) [same]

-------------------- testing n = 17

sortf(L,n):
  integer  3
  integer  4
  list     1 5 9 
  list     5 &null 5 
  list     4 4 4 4 
  list     3 3 3 
  list     &null 25 
  list     2 2 
  list     1 
  list     &null &null 
  list     
  list     1 2 3 4 5 95 96 97 98 99 100 
  r1       2 
  r1       4 
  r1       8 
  r3       3 1 4 
  r3       2 6 5 
  r3       3 5 &null 
  r3       7 8 9 

sortf(S,n) [same]

-------------------- testing n = -4

sortf(L,n):
  integer  3
  integer  4
  list     1 5 9 
  list     5 &null 5 
  list     3 3 3 
  list     &null 25 
  list     2 2 
  list     1 
  list     &null &null 
  list     
  list     4 4 4 4 
  list     1 2 3 4 5 95 96 97 98 99 100 
  r1       2 
  r1       4 
  r1       8 
  r3       3 1 4 
  r3       2 6 5 
  r3       3 5 &null 
  r3       7 8 9 

sortf(S,n) [same]

-------------------- testing n = -3

sortf(L,n):
  integer  3
  integer  4
  list     &null 25 
  list     2 2 
  list     1 
  list     &null &null 
  list     
  list     1 5 9 
  list     3 3 3 
  list     4 4 4 4 
  list     5 &null 5 
  list     1 2 3 4 5 95 96 97 98 99 100 
  r1       2 
  r1       4 
  r1       8 
  r3       2 6 5 
  r3       3 1 4 
  r3       3 5 &null 
  r3       7 8 9 

sortf(S,n) [same]

-------------------- testing n = -2

sortf(L,n):
  integer  3
  integer  4
  list     1 
  list     
  list     5 &null 5 
  list     &null 25 
  list     &null &null 
  list     2 2 
  list     3 3 3 
  list     4 4 4 4 
  list     1 5 9 
  list     1 2 3 4 5 95 96 97 98 99 100 
  r1       2 
  r1       4 
  r1       8 
  r3       3 1 4 
  r3       3 5 &null 
  r3       2 6 5 
  r3       7 8 9 

sortf(S,n) [same]

-------------------- testing n = -1

sortf(L,n):
  integer  3
  integer  4
  list     
  list     &null &null 
  list     1 
  list     2 2 
  list     3 3 3 
  list     4 4 4 4 
  list     5 &null 5 
  list     1 5 9 
  list     &null 25 
  list     1 2 3 4 5 95 96 97 98 99 100 
  r3       3 5 &null 
  r1       2 
  r1       4 
  r3       3 1 4 
  r3       2 6 5 
  r1       8 
  r3       7 8 9 

sortf(S,n) [same]
0
-1
    i        j       ~j      i & j    i | j    i ^ j   i << j   i >> j
       1        0
00000001 00000000 FFFFFFFF 00000000 00000001 00000001 00000001 00000001 
       1        1
00000001 00000001 FFFFFFFE 00000001 00000001 00000000 00000002 00000000 
       1        2
00000001 00000002 FFFFFFFD 00000000 00000003 00000003 00000004 00000000 
       1        3
00000001 00000003 FFFFFFFC 00000001 00000003 00000002 00000008 00000000 
       1        4
00000001 00000004 FFFFFFFB 00000000 00000005 00000005 00000010 00000000 
       1      100
00000001 00000064 FFFFFF9B 00000000 00000065 00000065 00000000 00000000 
    i        j       ~j      i & j    i | j    i ^ j   i << j   i >> j
       2        0
00000002 00000000 FFFFFFFF 00000000 00000002 00000002 00000002 00000002 
       2        1
00000002 00000001 FFFFFFFE 00000000 00000003 00000003 00000004 00000001 
       2        2
00000002 00000002 FFFFFFFD 00000002 00000002 00000000 00000008 00000000 
       2        3
00000002 00000003 FFFFFFFC 00000002 00000003 00000001 00000010 00000000 
       2        4
00000002 00000004 FFFFFFFB 00000000 00000006 00000006 00000020 00000000 
       2      100
00000002 00000064 FFFFFF9B 00000000 00000066 00000066 00000000 00000000 
    i        j       ~j      i & j    i | j    i ^ j   i << j   i >> j
       3        0
00000003 00000000 FFFFFFFF 00000000 00000003 00000003 00000003 00000003 
       3        1
00000003 00000001 FFFFFFFE 00000001 00000003 00000002 00000006 00000001 
       3        2
00000003 00000002 FFFFFFFD 00000002 00000003 00000001 0000000C 00000000 
       3        3
00000003 00000003 FFFFFFFC 00000003 00000003 00000000 00000018 00000000 
       3        4
00000003 00000004 FFFFFFFB 00000000 00000007 00000007 00000030 00000000 
       3      100
00000003 00000064 FFFFFF9B 00000000 00000067 00000067 00000000 00000000 
file(concord.dat)
11
"eoptera, ("
21
21
1229
"\n"
1231
